lib/matplotlib/contour.py

```diff
@@ -1586,70 +1611,48 @@ def _initialize_x_y(self, z):
+
+            contour([X, Y,] Z, [levels], **kwargs)
+
         :func:`~matplotlib.pyplot.contour` and
         :func:`~matplotlib.pyplot.contourf` draw contour lines and
         filled contours, respectively.  Except as noted, function
         signatures and return values are the same for both versions.
 
-        :func:`~matplotlib.pyplot.contourf` differs from the MATLAB
-        version in that it does not draw the polygon edges.
-        To draw edges, add line contours with
-        calls to :func:`~matplotlib.pyplot.contour`.
-
-
-        Call signatures::
-
-          contour(Z)
-
-        make a contour plot of an array *Z*. The level values are chosen
-        automatically.
-
-        ::
-
-          contour(X,Y,Z)
-
-        *X*, *Y* specify the (x, y) coordinates of the surface
-
-        ::
-
-          contour(Z,N)
-          contour(X,Y,Z,N)
-
-        contour up to *N+1* automatically chosen contour levels
-        (*N* intervals).
-
-        ::
 
-          contour(Z,V)
-          contour(X,Y,Z,V)
-
-        draw contour lines at the values specified in sequence *V*,
-        which must be in increasing order.
-
-        ::
-
-          contourf(..., V)
-
-        fill the ``len(V)-1`` regions between the values in *V*,
-        which must be in increasing order.
-
-        ::
-
-          contour(Z, **kwargs)
-
-        Use keyword args to control colors, linewidth, origin, cmap ... see
-        below for more details.
-
-        *X* and *Y* must both be 2-D with the same shape as *Z*, or they
-        must both be 1-D such that ``len(X)`` is the number of columns in
-        *Z* and ``len(Y)`` is the number of rows in *Z*.
-
-        ``C = contour(...)`` returns a
-        :class:`~matplotlib.contour.QuadContourSet` object.
+        Parameters
+        ----------
+        X, Y : array-like, optional
+            The coordinates of the values in *Z*.
+            
+            *X* and *Y* must both be 2-D with the same shape as *Z* (e.g.
+            created via :func:`numpy.meshgrid`), or they must both be 1-D such
+            that ``len(X) == M`` is the number of columns in *Z* and
+            ``len(Y) == N`` is the number of rows in *Z*.
+            
+            If not given, they are assumed to be integer indices, i.e.
+            ``X = range(M)``, ``Y = range(N)``. 
+
+        Z : array-like(N, M)
+            The height values over which the contour is drawn.
+        
+        levels : int or array-like, optional
+            Determines the number and positions of the contour lines / regions.
+            
+            If an int *n*, use *n* data intervals; i.e. draw *n+1* contour
+            lines. The level heights are automatically chosen.
+            
+            If array-like, draw contour lines the specified level values. 
+            The values must be in increasing order.
 
-        Optional keyword arguments:
+        Returns
+        -------
+        `matplotlib.contour.QuadContourSet`
```

Comment:
This should actually link to the QuadContourSet docs, 
<pre>
:class:`~matplotlib.contour.QuadContourSet` object. 
</pre>

(modulo correct syntax)